### ISSUE TITLE
[Bug]: [JSON] Search becomes slowly when searching after flush and int & json mixed expressions 

### ISSUE THREAD

**Issue by binbinlv** [milvus-io/milvus#24186]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master-20230515-30415e1b
- Deployment mode(standalone or cluster):both
- MQ type(rocksmq, pulsar or kafka):    all
- SDK version(e.g. pymilvus v2.0.0rc2):2.4.0.dev31
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Search time increased when searching with int and json expressions after flush
from: 0.16s to 35s

```
>>> t0 = time.time()
>>> res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "json_field[\"number\"] >= 0 && json_field[\"number\"] <= 10")
>>> tt = time.time() - t0
>>> print(tt)
0.16013193130493164
>>> collection.flush()
>>> res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "int64 <= 5 && json_field[\"number\"] >= 0 && json_field_1[\"number\"] <= 10")
>>> t0 = time.time()
>>> res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "json_field[\"number\"] >= 0 && json_field[\"number\"] <= 10")
>>> tt = time.time() - t0
>>> print(tt)
35.780673027038574
```

### Expected Behavior

The time consumption should be similar 

### Steps To Reproduce

```python
from pymilvus import CollectionSchema, FieldSchema
from pymilvus import Collection
from pymilvus import connections
from pymilvus import DataType
from pymilvus import Partition
from pymilvus import utility
import time

connections.connect()

dim = 128
int64_field = FieldSchema(name="int64", dtype=DataType.INT64, is_primary=True)
float_field = FieldSchema(name="float", dtype=DataType.FLOAT)
bool_field = FieldSchema(name="bool", dtype=DataType.BOOL)
string_field = FieldSchema(name="string", dtype=DataType.VARCHAR, max_length=65535)
json_field = FieldSchema(name="json_field", dtype=DataType.JSON)
json_field_1 = FieldSchema(name="json_field_1", dtype=DataType.JSON)
float_vector = FieldSchema(name="float_vector", dtype=DataType.FLOAT_VECTOR, dim=dim)
schema = CollectionSchema(fields=[int64_field, float_field, bool_field, json_field, json_field_1, float_vector])
collection = Collection("test_search_collection_multiple_json_0", schema=schema)
import random
nb=3000
vectors = [[random.random() for _ in range(dim)] for _ in range(nb)]
import numpy as np
res = collection.insert([[i for i in range(nb)], [np.float32(i) for i in range(nb)], [np.bool_(i) for i in range(nb)], [{"number": i, "string": "milvus", "bool": True, "list": [j for j in range(0, i)], "json": {"number": i, "string": "milvus", "bool": True}} for i in range(0, nb)], [{"number": i, "string": "milvus", "bool": True, "list": [j for j in range(0, i)], "json": {"number": i, "string": "milvus", "bool": True}} for i in range(0, nb)], vectors])
index_param = {"index_type": "IVF_SQ8", "metric_type": "L2", "params": {"nlist": 8192}}
collection.create_index("float_vector", index_param, index_name="index_name_1")
collection.load()
default_search_params = {"metric_type": "L2", "params": {"nprobe": 10}}
limit = 10
nq =1
t0 = time.time()
res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "json_field[\"number\"] >= 0 && json_field[\"number\"] <= 10")
tt = time.time() - t0
print(tt)
collection.flush()
res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "int64 <= 5 && json_field[\"number\"] >= 0 && json_field_1[\"number\"] <= 10")
t0 = time.time()
res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "json_field[\"number\"] >= 0 && json_field[\"number\"] <= 10")
tt = time.time() - t0
print(tt)
```

**Comment by binbinlv** [comment]
Verified and fixed in master branch:
pymilvus: 2.4.0.dev40
milvus: [master-20230522-7e9ef36d](https://hub.docker.com/layers/milvusdb/milvus/master-20230522-7e9ef36d/images/sha256-c691a2cf499765aae3386257713d791d9e2d3e3f53bff8616c7688ec2eeec4e0?context=explore)

```
>>> t0 = time.time()
>>> res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "json_field[\"number\"] >= 0 && json_field[\"number\"] <= 10")
>>> tt = time.time() - t0
>>> print(tt)
0.13869976997375488
>>> collection.flush()
>>> res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "int64 <= 5 && json_field[\"number\"] >= 0 && json_field_1[\"number\"] <= 10")
>>>
>>> t0 = time.time()
>>> res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "json_field[\"number\"] >= 0 && json_field[\"number\"] <= 10")
>>> tt = time.time() - t0
>>> print(tt)
0.13328003883361816
```

**Comment by binbinlv** [comment]
Verified and fixed in 2.2.0 branch:
pymilvus: 2.2.9.dev14
milvus: [2.2.0-20230524-70d5d9c1](https://hub.docker.com/layers/milvusdb/milvus/2.2.0-20230524-70d5d9c1/images/sha256-18651b0f1abc6406a061bf33616fc88dac91be733ccbd9fc8b0a196c38be8809?context=explore)

Results:
```
>>> t0 = time.time()
>>> res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "json_field[\"number\"] >= 0 && json_field[\"number\"] <= 10")
>>> tt = time.time() - t0
>>> print(tt)
0.13849401473999023
>>> collection.flush()
>>>
>>> res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "int64 <= 5 && json_field[\"number\"] >= 0 && json_field_1[\"number\"] <= 10")
>>>
>>> res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "int64 <= 5 && json_field[\"number\"] >= 0 && json_field_1[\"number\"] <= 10")
>>>
>>>
>>> t0 = time.time()
>>> res = collection.search(vectors[:nq], "float_vector", default_search_params, limit, "json_field[\"number\"] >= 0 && json_field[\"number\"] <= 10")
>>> tt = time.time() - t0
>>> print(tt)
0.1100010871887207
```
